molecularnodes/utils.py

```diff
@@ -40,6 +45,37 @@ def correct_periodic_positions(
     return final_positions
 
 
+def frame_mapper(
+    frame: int,
+    subframes: int = 0,
+    offset: int = 0,
+    mapping: np.ndarray = None,
+) -> list:
+    frame = max(frame - offset, 0)
+
+    if mapping is not None:
+        if not isinstance(mapping, np.ndarray):
+            raise ValueError(
+                "Frame mapping must be an array of values to map frames to"
+            )
+        # add the subframes to the frame mapping
+        frame_map = np.repeat(mapping, subframes + 1)
+        # get the current and next frames
+        frame_a = frame_map[frame]
+
+    frame_a = frame
+
+    if subframes > 0:
+        frame_a = int(frame / (subframes + 1))
+
+    return frame_a
+
+
+def frames_to_average(frame: int, average: int = 0) -> list[int]:
```

Comment:
When `frame=0`, this will also generate negative values; maybe better just filtering those values out?

Reply:
good idea, added it in